src/lib/falcor/routes/articles/by-slug.js

```diff
@@ -171,6 +198,71 @@ export default [
       });
     },
   },
+  {
+    // Add tags to an article
+    route: "articles['bySlug'][{keys:slugs}]['tags']['updateTags']",
+    call: (callPath, args) => {
+      // the falcor.model.call only takes a path not a pathset
+      // so it is not possible to call this function for more
+      // than 1 article at a time, therefore we know keys:slugs is length 1
+      if (callPath.slugs > 1) {
+        throw new Error(
+          'updateTags falcor function was called illegally with more than 1 article slug',
+        );
+      }
+      return new Promise(async resolve => {
+        const articleId = args[0];
+        const tags = args[1];
+        const articleSlug = callPath.slugs[0];
+
+        const newTags = tags.filter(tagObject => tagObject.id === null);
+        const createPromises = newTags.map(tagObject => {
+          const filteredTagObject = _.omit(tagObject, 'id');
+          return serverModel.call(
```

Comment:
I'm not a big fan of calling another falcor call inside a falcor call, I would imagine that you could just call a DB function that creates all the needed tags and updates them and returns an array of all new tags created and another array (both arrays wrapped in a top level object of some kind) that is the new tags added to the article, and then you just make pathsets of that, since it's a call you can return any path sets you want, such as new tags, under the `tags` route

Reply:
You could also call several DB functions, but I'd much prefer calling database functions directly here instead of calling falcor APIs.

... Or is your argument that this takes advantage of caching?

Reply:
@emilgoldsmith I thought about doing that. My reason for choosing this way though was less due to the advantages of caching and more so in order not to duplicate code. In case I do use the db calls directly, I will have to clean up the json-graph which is already done by the falcor call. Therefore, though not ideal, I chose to go with the extra, ugly falcor call instead. LMKWYT.

Reply:
There may be an argument for caching, but otherwise I'd definitely much rather just call db functions, which I think could still easily be DRY, but given that I don't see anything else dodgy in a quick look through now I think we can merge it in and maybe just make an issue for looking at this again, as I do think it would be nicer code without these recursive calls